src/submit-modal/submit-modal.js

```diff
@@ -0,0 +1,275 @@
+    <>
+        <ModalContent>
+            <Warning
+                error={true}
+                title={i18n.t('There was a problem submitting data')}
+            >
+                <span>{error.message}</span>
+            </Warning>
+        </ModalContent>
+
+        <ModalActions>
+            <ButtonStrip end>
+                <Button onClick={onClose}>{i18n.t('Close')}</Button>
+                <Button primary onClick={onRetry}>
+                    {i18n.t('Try again')}
+                </Button>
+            </ButtonStrip>
+        </ModalActions>
+    </>
+)
+
+ErrorModalContent.propTypes = {
+    error: PropTypes.object,
+    onClose: PropTypes.func,
+    onRetry: PropTypes.func,
+}
+
+const SuccessModalContent = ({ onClose, data }) => (
+    <>
+        <ModalContent>
+            <SuccessContent data={data} />
+        </ModalContent>
+
+        <ModalActions>
+            <ButtonStrip end>
+                <Button onClick={onClose}>{i18n.t('Close')}</Button>
+            </ButtonStrip>
+        </ModalActions>
+    </>
+)
+
+SuccessModalContent.propTypes = {
+    data: PropTypes.object,
+    onClose: PropTypes.func,
+}
+
+const getReportText = (request) => {
+    const { name, orgUnits, periods } = request
+    const orgUnitText =
+        orgUnits.length === 1
+            ? i18n.t('{{orgUnitCount}} organisation unit', {
+                  orgUnitCount: orgUnits.length,
+              })
+            : i18n.t('{{orgUnitCount}} organisation units', {
+                  orgUnitCount: orgUnits.length,
+              })
+    let periodsText = ''
+    if (periods.length === 1) {
+        periodsText = i18n.t('{{periodsCount}} period: {{periods}}', {
+            periodsCount: periods.length,
+            periods: periods[0],
+            nsSeparator: '-:-',
+        })
+    }
+    if (periods.length > 1 && periods.length <= 3) {
+        periodsText = i18n.t('{{periodsCount}} periods: {{periods}}', {
+            periodsCount: periods.length,
+            periods: periods.join(', '),
+            nsSeparator: '-:-',
+        })
+    }
+    if (periods.length > 3) {
+        periodsText = i18n.t(
+            '3+ periods: {{periods}}, and {{periodsCountLessThree}} more',
+            {
+                periodsCount: periods.length,
+                periods: periods.slice(0, 3).join(', '),
+                periodsCountLessThree: periods.length - 3,
+                nsSeparator: '-:-',
+            }
+        )
+    }
+    return `${name}, ${orgUnitText}, ${periodsText}`
+}
+
+const ConfirmModalContent = ({ exchange, requests, onClose, onSubmit }) => {
+    // this is very wordy, but did not have luck with i18nextscanner picking up from more compact versions...
+    let reportTranslationsString
+    const reportCount = requests.length
+    const exchangeName = exchange?.displayName
+    const exchangeURL = exchange?.target?.api?.url
+
+    if (exchange?.target?.type === 'INTERNAL') {
```

Comment:
discussed with Joe. If exchange is internal we should have this be '{{reportCount}} reports to {{-exchangeName}} internally at {{-exchangeURL}}'